source/extensions/filters/http/ext_authz/ext_authz.cc

```diff
@@ -212,8 +212,10 @@ void Filter::onComplete(Filters::Common::ExtAuthz::ResponsePtr&& response) {
           ENVOY_STREAM_LOG(trace,
                            "ext_authz filter added header(s) to the local response:", callbacks);
           for (const auto& header : headers) {
-            ENVOY_STREAM_LOG(trace, " '{}':'{}'", callbacks, header.first.get(), header.second);
             response_headers.remove(header.first);
+          }
+          for (const auto& header : headers) {
```

Comment:
Yeah, the requirement of these two-passes is not obvious at first glance. Could you add comments around this? Transferring the proposed solution text in https://github.com/envoyproxy/envoy/issues/8649 should be good. But whatever explains why we're doing this helps, up to you.

Reply:
Thanks for the feedback, @dio. I pushed another commit to add the suggested comments. Please take a look.